libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -41,6 +42,7 @@ type
     triggerSelf*: bool                # trigger own local handler on publish
     cleanupLock: AsyncLock
     validators*: Table[string, HashSet[ValidatorHandler]]
+    observers: ref seq[PubSubObserver] # ref as in smart_ptr
```

Comment:
Why we need here reference to sequence, if `PubSub` is already reference object?

Reply:
As the comment says as `smart_ptr` because it is shared in the Peers